src/test/java/com/google/api/generator/engine/ast/WhileStatementTest.java

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Test;
+
+public class WhileStatementTest {
+  @Test
+  public void validWhileStatement_simple() {
+    WhileStatement whileStatement =
+        WhileStatement.builder()
+            .setConditionExpr(createConditionExpr("condition"))
+            .setBody(createAssignmentExprList())
+            .build();
+    assertTrue(whileStatement.conditionExpr().type().equals(TypeNode.BOOLEAN));
+  }
+
+  @Test
+  public void validWhileStatement_booleanObjectCondition() {
+    // The condition expression type can be boolean or its boxed type.
+    VariableExpr condExpr =
+        VariableExpr.withVariable(
+            Variable.builder().setName("condition").setType(TypeNode.BOOLEAN_OBJECT).build());
+    WhileStatement whileStatement =
+        WhileStatement.builder()
+            .setConditionExpr(condExpr)
+            .setBody(createAssignmentExprList())
+            .build();
+    assertTrue(whileStatement.conditionExpr().type().equals(TypeNode.BOOLEAN));
```

Comment:
`assertEquals`? Same elsewhere.

Reply:
This assertTrue is intended, since it does not have to be `boolean`, this check should also allow `Boolean` object, same elsewhere, let me know which one you prefer. And I will make it consistent. Thanks

Reply:
Why wouldn't `assertEquals(TypeNode.BOOLEAN, whileStatement.conditionExpr().type())` check for `Boolean`?

Reply:
Got the point, fixed it.